### ISSUE TITLE
Inconsistent package contents dependent on build system

### ISSUE THREAD

**Issue by AMDmi3** [open-source-parsers/jsoncpp#1170]
**Describe the bug**
Using different build systems lead to different package contents:
* Meson
  - no CMake modules are generated
  - both shared and static libraries are possible with `--default-library=both`
* CMake:
  - CMake modules are generated
  - only either of shared or static libraries is possible
* additionally, shared libarary is installed as `libjsoncpp.so.1.9.2` with cmake and `libjsoncpp.so.22` with meson

This makes it much harder to use jsoncpp, as jsoncpp consumer may not rely on any inconsistent component present. For instance, a project which relies on CMake modules will fail to build on FreeBSD which uses meson to build jsoncpp.

**To Reproduce**
Build, install with cmake and meson, compare produced files

**Expected behavior**
Single build system, or consistent behavior with both build systems. At the very least, a way to build all components which consumers may requiered (cmake modules AND static library AND shared library).

**Comment by mr700** [comment]
The only static/shared library problem with cmake can be solved by calling cmake two times and packaging the relevant files.

**Comment by AMDmi3** [comment]
> The only static/shared library problem with cmake can be solved by calling cmake two times and packaging the relevant files.

Not really, this is usually solved with two separate targets for static and shared library.

**Comment by dota17** [comment]
All the cmake support is user-provided, not official. We actually use meson in Travis CI.

> At the very least, a way to build all components which consumers may requiered (cmake modules AND static library AND shared library).

Welcome a PR.

**Comment by dota17** [comment]
Related PR: #827 

I found that before this PR,  there are different targets for static and shared library. And the target name wasn't unique. Also, you could find these target in `1.8.4` or `0.10.7`.

> Not really, this is usually solved with two separate targets for static and shared library.

Thus, if we want to change the targets, please consider the influences of #827.
What is more, could we add [`OBJECT LIBRARY`](https://cmake.org/cmake/help/v3.8/command/add_library.html#object-libraries) except static and shared library?
